### PR TITLE
feat: Add overloaded version for Insert() & Delete() in common_checker.h

### PR DESCRIPTION
**Pull request by xzhseh**
This PR introduces overloaded versions of the `Insert()` and `Delete()` functions in `common_checker.h`

The original versions of these functions were specifically tailored to comply with the Gradescope tests, which made them somewhat inflexible. 
To enhance usability, the newly added overloaded versions support a broader range of input formats.

The main objective of these additions is to facilitate students in writing their own test cases. 
It will allow them to thoroughly test their implementation locally, providing a more flexible and user-friendly testing environment.

This update does not impact the original functions, ensuring continued compatibility with Gradescope tests.

### CODE REVIEW

File: test/concurrency/common_checker.h [hunk 1]
```diff
@@ -39,6 +100,31 @@ auto Delete(Transaction *txn, BustubInstance &instance, int v1) -> void {
...
+
+  assert(!d_vec.empty());
+
+  std::stringstream ss;
+  auto writer = bustub::SimpleStreamWriter(ss, true, ",");
+
+  for (const auto &v1 : d_vec) {
+    fmt::print(stderr, "delete data with v1 = {} in txn {} {}\n", v1, txn->GetTransactionId(), txn->GetIsolationLevel());
+    std::string sql = fmt::format("DELETE FROM t1 WHERE v1 = {}", v1);
+    instance.ExecuteSqlTxn(sql, writer, txn);
```
Comment (skyzh) [inline test/concurrency/common_checker.h:122]: check return value == true otherwise print a warning

File: test/concurrency/common_checker.h [hunk 2]
```diff
@@ -30,6 +30,67 @@ auto Insert(Transaction *txn, BustubInstance &instance, int v1) -> void {
...
+
+  assert(v1_vec.size() == 1);
+  std::string val_str{};
+  if (!flag) {
+    for (size_t i = 0; i < v1_vec.size(); ++i) {
+      val_str += fmt::format("({}, {}), ", v1_vec[i], v2_vec[i]);
+    }
+  } else {
+    for (const auto &v2 : v2_vec) {
+      val_str += fmt::format("({}, {}), ", v1_vec[0], v2);
```
Comment (skyzh) [inline test/concurrency/common_checker.h:75]: does the parser work with trailing comma?

**Review by skyzh — COMMENTED** [review]
rest LGTM, good work!

File: test/concurrency/common_checker.h (hunk 2 above)
Comment (xzhseh) [reply test/concurrency/common_checker.h:75]: I removed the trailing comma and space on line 81 & 82, so I guessed it should be good.
But though, I tested `libpg_query` locally and found out it supported neither parsing trailing comma nor (trailing comma + extra space)
image

File: test/concurrency/common_checker.h (hunk 1 above)
Comment (xzhseh) [reply test/concurrency/common_checker.h]: Sure, I also changed the sequence of printing messages (&& add result checker) for `Insert()`

**Comment by xzhseh** [comment]
Seems that the format is not consistent with the format specified by running `make format`, I'll run it later at noon, since I don't have the neccessary packages installed on the current machine :(

**Review by skyzh — APPROVED** [review]

**Pull request merged**